mucommander-core/src/main/java/com/mucommander/ui/action/impl/ToggleTerminalAction.java

```diff
@@ -36,6 +36,13 @@
  */
 public class ToggleTerminalAction extends ActiveTabAction {
 
+    /**
+     * A reference to Terminal Integration. Made static (and volatile) to ensure
+     * that TerminalIntegration is initialized only once, otherwise, for some unknown
+     * to me reason even-though ToggleTerminalAction constructor is called once, the
+     * TerminalIntegration is called twice as if there was another instance of
+     * ToggleTerminalAction with terminalIntegration = null - but there's only one instance (!).
+     */
     private volatile static TerminalIntegration terminalIntegration = null;
```

Comment:
```suggestion
    private volatile static TerminalIntegration terminalIntegration;
```
while on it.. and what does `the TerminalIntegration is called twice` mean, that the constructor of `TerminalIntegration` is called twice?

Reply:
> and what does `the TerminalIntegration is called twice` mean, that the constructor of `TerminalIntegration` is called twice?

ah ok, [you replied on that already](https://github.com/mucommander/mucommander/pull/1190#discussion_r1600694968)
interesting..

Reply:
yup, the constructor of TerminalIntegration. Going to add 'constructor' to that comment.

Reply:
ok, so I took a further look at it and I see that the constructor of TerminalIntegration is called from three places during startup.

the thread that initializes the toolbar:
![image](https://github.com/mucommander/mucommander/assets/6020068/020c5f1c-15ba-4670-8a20-967e33a1c692)

the thread that initializes the menubar:
![image](https://github.com/mucommander/mucommander/assets/6020068/765fdb2a-6e5c-48c4-9f80-842ac23077db)

and the thread that initializes the mainframe (MainFrameInit):
![image](https://github.com/mucommander/mucommander/assets/6020068/9f8bd23a-b67a-4ecb-abdc-144c83a414f6)

setting `terminalIntegration` as static prevents its instantiation several times but it seems problematic to have it static and initializing it with a main frame, what happens with several main frames?.. I think it would be better to "inject" the TerminalIntegration from the MainFrame

Reply:
@pskowronek could you please review the changes in the last two commits?

Reply:
nice catch! So it was rather a problem with catching breakpoints in multithreading?

> could you please review the changes in the last two commits?
checking...

Reply:
> nice catch! So it was rather a problem with catching breakpoints in multithreading?

yeah, probably

Reply:
on Eclipse I haven't had such problems, but maybe under Idea there are some caveats I don't get.
It was probably a problem with half-baked class initialized in the same time, nevertheless - interesting.